src/modules/hooks/useBlockExplorer.test.ts

```diff
@@ -0,0 +1,53 @@
+import { renderHook } from '@testing-library/react';
+import { useChains } from 'wagmi';
+import { useBlockExplorer } from './useBlockExplorer';
+
+jest.mock('wagmi', () => ({
+    useChains: jest.fn(),
+}));
+
+describe('useBlockExplorer', () => {
+    const mockChains = [
+        {
+            id: 1,
+            name: 'Ethereum',
+            blockExplorers: { default: { url: 'https://etherscan.io' } },
+        },
+        {
+            id: 137,
+            name: 'Polygon',
+            blockExplorers: { default: { url: 'https://polygonscan.com' } },
+        },
+    ];
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('generates correct URL for different entity types and chain IDs', () => {
+        (useChains as jest.Mock).mockReturnValue(mockChains);
```

Comment:
If we use `jest.spyOn` we don't need to add such casts

Reply:
✅ https://github.com/aragon/ods/pull/220/commits/7cea3eb11722970012e7282e7d86a963d3861f5b